src/btree/tuple_chunk.c

```diff
@@ -0,0 +1,521 @@
+/*-------------------------------------------------------------------------
+ *
+ * tuple_chunk.c
+ *		OrioleDB implementation of chunk API over tuple chunks.
+ *
+ * Copyright (c) 2024, Oriole DB Inc.
+ *
+ * IDENTIFICATION
+ *	  contrib/orioledb/include/btree/tuple_chunk.c
+ *
+ *-------------------------------------------------------------------------
+ */
+#include "postgres.h"
+
+#include "btree/btree.h"
+
+#include "btree/chunk_ops.h"
+#include "btree/find.h"
+#include "btree/tuple_chunk.h"
+
+static void insert_chunk_item(BTreeChunkDesc *desc, OffsetNumber itemOffset,
+							  uint16 itemSize);
+static void resize_chunk_item(BTreeChunkDesc *desc, OffsetNumber itemOffset,
+							  uint16 newItemSize);
+static void delete_chunk_item(BTreeChunkDesc *desc, OffsetNumber itemOffset);
+
+static bool load_chunk_partially(BTreeChunkDesc *desc, PartialPageState *partial,
+								 Page page);
+
+static uint16 get_tuple_chunk_header_size(BTreeChunkType chunkType);
+static uint16 get_tuple_chunk_tuple_size(BTreeDescr *treeDesc,
+										 BTreeChunkType chunkType, OTuple tuple);
+
+static void add_builder_item(BTreeChunkBuilderData *builderData,
+							 Pointer tupleHeader, uint16 tupleHeaderSize,
+							 OTuple tuple, uint16 tupleSize);
+
+int32
+tuple_chunk_estimate_change(BTreeChunkDesc *desc, OffsetNumber itemOffset,
```

Comment:
We need to also estimate the size of the new BTreePageChunk.items element.

Reply:
Thanks. I added the size of LocationIndex shift to the calculation.